master/pkg/actor/ref.go

```diff
@@ -83,6 +83,20 @@ func (r *Ref) Parent() *Ref {
 	return r.parent
 }
 
+// Children returns a list of references to the actor's children.
+func (r *Ref) Children() []*Ref {
+	children := make([]*Ref, 0, len(r.children))
+	for _, child := range r.children {
+		children = append(children, child)
+	}
+	return children
```

Comment:
I can't imagine a place where the caller should mutate `r.children`. Is there an example of where this is needed in this PR, I couldn't find one?

Reply:
There is no such example in our codebase because we have a history of writing good code. :) But technically we could mutate and I don't see a reason why should remove this copy process. This is also how we implement `Context.Children()`. If you think it's an inefficient code, we probably should remove the copying process in both places.

Reply:
Ah, missed this was on the `*Ref` at first. I didn't see this used in the PR, but I saw a `ctx.Children()` so I got confused. More of a meta-question: should anything ever be doing this, or just sending a message to the `Ref` to forward to all its children?

Reply:
As for the copying, I don't think there's any way to do it.

> But technically we could mutate and I don't see a reason why should remove this copy process.

Since `children` is unexported I think we're good on this.

Reply:
Really don't have any issue with this, more or less just curious. Just reviewed because I was a "code owner" for the agent stuff and realized I gave you conflicts on this stuff.

Reply:
> Ah, missed this was on the *Ref at first. I didn't see this used in the PR, but I saw a ctx.Children() so I got confused

Yeah. I used that in a previous version but now I don't use `Ref.Children()` at all. I'll probably remove this change.

>  More of a meta-question: should anything ever be doing this, or just sending a message to the Ref to forward to all its children?

In the case of resource pools, we only allow one resource pool and we simply ignore the message if a resource pool finds out the message it receives is about a task it doesn't own. The benefit of designing this way is that the agent resource manager would be very simple and not need to keep track of tasks, which might introduce the complexity of syncing the tasks between the agent resource manager and resource pools. We might need to use this complex approach in the future if we want one task to be able to use multiple resource pools.

> Since children is unexported I think we're good on this (that is, if we just didn't have this at all).
> Really don't have any issue with this, more or less just curious. Just reviewed because I was a "code owner" for the agent stuff and realized I gave you conflicts on this stuff.

Thanks for reviewing this anyway. I am fine with either way in this PR. But I'd probably not change the style `actor` has in this PR unless there is a necessary reason to do that.